### ISSUE TITLE
NS7 migration nativescript-animated-circle issue in 'setFillCircleColor'

### ISSUE THREAD

**Issue by llpl** [NativeScript/plugins#55]
We were using nativescript-animated-circle and now I am trying to upgrade our project NS7 for IOS 14.1 crash issue.
Now I tried to use @nativescript/animated-circle 1.1.5 but while running for android it is giving error 

 > Cannot read property 'setFillCircleColor' of undefined

because of this plugin If you see plugin code ./node_modules/@nativescript/animated-circle/index.android.js has the reference for 

 set fillColor(value) {
        this._fillColor = value;
        if (value instanceof Color) {
            this.android.setFillCircleColor(value.argb);
        }
        else {
            this.android.setFillCircleColor(new Color(value).argb);
        }
    }
----------------------------------------------------------------------------------------------------------------------------
My project configuration :

——————————————————————————————————————————————————————————————

┌─────────────────────────────────────────────────────────┬─────────┐
│ Plugin                                                  │ Version │
│ @nativescript-community/ui-material-bottomnavigationbar │ ^5.1.16 │
│ @nativescript/animated-circle                           │ ^1.1.5  │
│ @nativescript/appversion                                │ ^2.0.0  │
│ @nativescript/core                                      │ ^7.0.13 │
│ @nativescript/local-notifications                       │ ^5.1.0  │
│ @nota/nativescript-webview-ext                          │ ^7.0.1  │
│ @nstudio/nativescript-floatingactionbutton              │ ^3.0.4  │
│ @nstudio/nativescript-pulltorefresh                     │ ^3.0.1  │
│ @proplugins/nativescript-cardview                       │ ^5.0.0  │
│ @proplugins/nativescript-dom                            │ ^2.2.3  │
│ @proplugins/nativescript-globalevents                   │ ^1.4.4  │
│ @proplugins/nativescript-orientation                    │ ^2.5.0  │
│ @proplugins/nativescript-pager                          │ ^12.0.3 │
│ @proplugins/nativescript-purchase                       │ ^6.0.0  │
│ crypto-js                                               │ ^4.0.0  │
│ fuse.js                                                 │ ^3.4.6  │
│ nativescript-feedback                                   │ ^2.0.0  │
│ nativescript-https                                      │ ^2.2.2  │
│ nativescript-intl                                       │ ^4.0.2  │
│ nativescript-iqkeyboardmanager                          │ ^1.5.1  │
│ nativescript-nodeify                                    │ ^0.8.0  │
│ nativescript-randombytes                                │ ^1.0.0  │
│ nativescript-sqlite                                     │ ^2.6.6  │
│ nativescript-toasts                                     │ ^1.0.3  │
│ sjcl                                                    │ ^1.0.8  │
│ uuid                                                    │ ^3.3.3  │
└─────────────────────────────────────────────────────────┴─────────┘
Dev Dependencies:
┌────────────────────────────┬─────────┐
│ Plugin                     │ Version │
│ @nativescript/android      │ 7.0.1   │
│ @nativescript/ios          │ 7.0.6   │
│ @nativescript/webpack      │ ~3.0.0  │
│ nativescript-worker-loader │ ^0.12.1 │
└────────────────────────────┴─────────┘

**Comment by senner008** [comment]
It's not working on Android either

**Comment by senner008** [comment]
Actually I can not set any of the settings

**Comment by 7ammer** [comment]
Fails in iOS, colour issues on android :(

I'm using nativescript-vue

**Comment by senner008** [comment]
I'm using nativescript 7 with angular 11.

The old one worked flawlessly :
https://github.com/sean-perkins/nativescript-animated-circle

It's not really doing anything other than extending the native circle view component.
Isn't it just a matter of doing a copy paste and updating the depencies for ns 7?

**Comment by senner008** [comment]
I can see that it works with NS7 Android with the old code if the dependencies are updated:

1. Install the new plugin 
2. Create 'animated-circle.ts'
3. Register the view like this:

```
import { registerElement } from '@nativescript/angular'
import { AnimatedCircle } from '@src/app/components/bluetooth-operations/animated-circle';

registerElement('AnimatedCircle', () => AnimatedCircle);
```
 'animated-circle.ts' :
```
// @ts-nocheck
import { Color, Utils } from '@nativescript/core';
import { ContentView } from '@nativescript/core/ui/content-view';

declare const at;

export class AnimatedCircle extends ContentView {
    private _android: any;
    private _progress: number;
    private _animateFrom: number;
    private _animationDuration = 1000;
    private _animated: boolean;
    private _maxValue = 100;
    private _rimColor = '#FF5722';
    private _rimWidth = 5;
    private _spinBarColor: any;
    private _barWidth: number;
    private _startAngle: number;
    private _text = '';
    private _textColor = new Color('orange');
    private _textSize = 28 * 10;

    clockwise = true;

    barColor = '#3D8FF4';
    fillColor: any;


    constructor() {
        super();
    }

    createNativeView() {
        return new at.grabner.circleprogress.CircleProgressView(this._context, null);
    }

    initNativeView() {
        this.android.setAutoTextSize(false);
        this.android.setTextMode(at.grabner.circleprogress.TextMode.TEXT);
        this.android.setTextScale(1.1);
        this.android.setTextSize(300);
        this.android.setUnitVisible(false);
        this.android.setOuterContourSize(0);
		this.android.setInnerContourSize(0);
        this.updateAnimatedCircle();
    }

    get android() {
        return this.nativeView;
    }

    set progress(value: number) {
        this._progress = Number(value);
        this.updateAnimatedCircle();
    }

    get progress(): number {
        return this._progress;
    }

    set animateFrom(value: number) {
        this._animateFrom = Number(value);
        this.updateAnimatedCircle();
    }

    get animateFrom(): number {
        return this._animateFrom;
    }

    set animationDuration(value: number) {
        this._animationDuration = Number(value);
        this.updateAnimatedCircle();
    }

    get animationDuration(): number {
        return this._animationDuration;
    }

    set animated(value: boolean) {
        this._animated = Boolean(value);
        this.updateAnimatedCircle();
    }

    get animated(): boolean {
        return this._animated;
    }

    set maxValue(value: number) {
        this._maxValue = Number(value);
        this.updateAnimatedCircle();
    }

    get maxValue(): number {
        return this._maxValue;
    }

    set rimColor(value: any) {
        this._rimColor = value;
        this.updateAnimatedCircle();
    }

    get rimColor() {
        return this._rimColor;
    }

    set rimWidth(value: number) {
        this._rimWidth = Number(value);
        this.updateAnimatedCircle();
    }

    get rimWidth() {
        return this._rimWidth;
    }

    set spinBarColor(value: any) {
        this._spinBarColor = value;
        this.updateAnimatedCircle();
    }

    get spinBarColor() {
        return this._spinBarColor;
    }

    set startAngle(value: number) {
        this._startAngle = Number(value);
        this.updateAnimatedCircle();
    }

    get startAngle() {
        return this._startAngle;
    }

    set barWidth(value: number) {
        this._barWidth = Number(value);
        this.updateAnimatedCircle();
    }

    get barWidth() {
        return this._barWidth;
    }

    set text(value: string) {
        this._text = value;
        this.updateAnimatedCircle();
    }

    get text() {
        return this.android.getText();
    }

    set textColor(value: string) {
        this._textColor = new Color(value);
        this.updateAnimatedCircle();
    }

    set textSize(value: number) {
        this._textSize = value * 10;
        this.updateAnimatedCircle();
    }

    get textSize() {

        return this.android.getTextSize();
    }

    private updateAnimatedCircle(): void {
        if (this.android) {
            this.android.setText(this._text);
            this.android.setTextColor(this._textColor.argb);
            this.android.setTextSize(this._textSize);
            if (this.animated) {
                if (this.animateFrom) {
                    this.android.setValueAnimated(this.animateFrom, this.progress, this.animationDuration);
                }
                else {
                    if (!this._animationDuration) {
                        this.android.setValueAnimated(this.progress);
                    }
                    else {
                        this.android.setValueAnimated(this.progress, this.animationDuration);
                    }
                }
            }
            else {
                this.android.setValue(this.progress);
            }
            this.android.setMaxValue(this.maxValue);
            if (this.rimColor) {
                this.android.setRimColor(new Color(this.rimColor).argb);
            }
            if (this.spinBarColor) {
                this.android.setSpinBarColor(new Color(this.spinBarColor).argb);
            }
            if (this.startAngle) {
                this.android.setStartAngle(this.startAngle);
            }
			if (this.barWidth) {
				this.android.setBarWidth(this.barWidth);
			} else {
				if (this.rimWidth) {
					// set rim width to bar width if no bar width provided
					this.android.setBarWidth(this.rimWidth);
				}
			}
            if (this.rimWidth) {
                this.android.setRimWidth(this.rimWidth);
            }
            if (this.barColor) {
                this.android.setBarColor([new Color(this.barColor).argb]);
            }
            if (this.fillColor) {
                this.android.setFillCircleColor(new Color(this.fillColor).argb);
            }

            this.android.setDirection(this.clockwise ? at.grabner.circleprogress.Direction.CW : at.grabner.circleprogress.Direction.CCW);
        }
    }

}
```

**Comment by NathanWalker** [comment]
This fix is published in `1.1.6` - let us know if encounter the same or further issue here.
